src/main/java/org/swisspush/redisques/action/DeleteAllQueueItemsAction.java

```diff
@@ -36,40 +37,33 @@ public void execute(Message<JsonObject> event) {
         p.onSuccess(redisAPI -> {
             redisAPI.del(Collections.singletonList(buildQueueKey(queue)), deleteReply -> {
                 if (deleteReply.failed()) {
-                    log.warn("Failed to deleteAllQueueItems. But we'll continue anyway",
-                            new Exception(deleteReply.cause()));
-                    // May we should 'fail()' here. But:
-                    // 1st: We don't, to keep backward compatibility
-                    // 2nd: We don't, to may unlock below.
+                    handleFail(event, "Operation DeleteAllQueueItems failed", deleteReply.cause());
+                    return;
```

Comment:
I would say yes because of these points:

- Only locks would remain. These can be deleted separately
- When already the deletion of the queue items fail, there is a high chance of also deleting the lock will fail
- When the queue items could not be deleted I would be more comfortable having the queue still locked